sdk/instrumentation/opentelemetry-instrumentation-azure-sdk/src/instrumenter.ts

```diff
@@ -21,11 +28,26 @@ export class OpenTelemetryInstrumenter implements Instrumenter {
     name: string,
     spanOptions: InstrumenterSpanOptions
   ): { span: TracingSpan; tracingContext: TracingContext } {
-    const span = trace
-      .getTracer(spanOptions.packageName, spanOptions.packageVersion)
-      .startSpan(name, toSpanOptions(spanOptions));
-
     let ctx = spanOptions?.tracingContext || context.active();
+    let span: Span;
+
+    if (environmentVariableToBoolean("AZURE_TRACING_DISABLED")) {
+      // disable only our spans but not any downstream spans
+      span = trace.wrapSpanContext(INVALID_SPAN_CONTEXT);
```

Comment:
nit: ideally this check should happen in core-tracing and applies to all potential tracing plugins

Reply:
I think that makes sense, but I am hesitant to add this to our core-tracing library as it will no-op by default anyway and I want to keep core-tracing's logic as narrow as possible. If I'm thinking about it correctly, adding this logic to core-tracing later if necessary is not a breaking change whereas removing it from core-tracing after we GA is. If that's right, maybe we can move that over if we ever have another plugin for tracing?